### PR TITLE
Smt int

### LINKED ISSUE #147: SMT Int class [Enhancement]
For building up constraints for synthesis problems it is useful to have access to an Int class that plays nicely with SMTBit.

Current implementation in branch: smt_int

Semantics defined in: https://smtlib.cs.uiowa.edu/theories-Ints.shtml

Would be nice to include methods to translate between SMTInt and SMTBitVector.

### PR DESCRIPTION
**Pull request by rdaly525**
Resolves #147

### CODE REVIEW

File: hwtypes/smt_bit_vector.py [hunk 1]
```diff
@@ -643,11 +643,13 @@ def substitute(self, *subs : tp.List[tp.Tuple["SBV", "SBV"]]):
 #    def __int__(self):
 #        return self.as_uint()
 #
-#    def as_uint(self):
-#        return self._value.bv_unsigned_value()
-#
-#    def as_sint(self):
-#        return self._value.bv_signed_value()
+    def as_uint(self):
```
Comment (cdonovick) [inline hwtypes/smt_bit_vector.py:646]: could you add a comment about avoiding circular imports?

File: hwtypes/smt_int.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        elif name is AUTOMATIC and value is SMYBOLIC:
+            name = _gen_name()
+            _name_table[name] = self
+
+        if value is SMYBOLIC:
+            self._value = smt.Symbol(name, INT)
+        elif isinstance(value, pysmt.fnode.FNode):
+            if value.get_type().is_int_type():
+                self._value = value
+            else:
```
Comment (cdonovick) [inline hwtypes/smt_int.py:79]: Maybe allow construction from pysmt BV?

**Review by cdonovick — APPROVED** [review]
Looks good modulo my comments

File: hwtypes/smt_bit_vector.py (hunk 1 above)
Comment (rdaly525) [reply hwtypes/smt_bit_vector.py:646]: Added comments

File: hwtypes/smt_int.py (hunk 2 above)
Comment (rdaly525) [reply hwtypes/smt_int.py:79]: Added the construction and a test

**Pull request merged**